WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java

```diff
@@ -87,48 +90,44 @@ public int onStartCommand(Intent intent, int flags, int startId) {
             return START_NOT_STICKY;
         }
 
-        StatsTimeframe period = StatsTimeframe.DAY;
+        final StatsTimeframe period;
         if (intent.hasExtra(ARG_PERIOD)) {
             period = (StatsTimeframe) intent.getSerializableExtra(ARG_PERIOD);
+        } else {
+            period = StatsTimeframe.DAY;
         }
 
-        String requestedDate = intent.getStringExtra(ARG_DATE);
-        if (requestedDate == null) {
+        final String requestedDate;
+        if (intent.getStringExtra(ARG_DATE) == null) {
             AppLog.w(T.STATS, "StatsService is started with a NULL date on this blogID - "
                     + blogId + ". Using current date!!!");
             int parsedBlogID = Integer.parseInt(blogId);
             int localTableBlogId = WordPress.wpDB.getLocalTableBlogIdForRemoteBlogId(parsedBlogID);
             requestedDate = StatsUtils.getCurrentDateTZ(localTableBlogId);
-        }
-
-        // True when the network call to update the graph is needed
-        boolean updateAlltimeStats = intent.getBooleanExtra(ARG_UPDATE_ALLTIME_STATS, true);
-        boolean updateGraphStats = intent.getBooleanExtra(ARG_UPDATE_GRAPH_STATS, true);
-
-        if (mServiceBlogId == null) {
-            startTasks(blogId, period, requestedDate, updateGraphStats, updateAlltimeStats, startId);
-        } else if (blogId.equals(mServiceBlogId) && mServiceRequestedTimeframe == period
-                && requestedDate.equals(mServiceRequestedDate)) {
-            // already running on the same blogID, same period
-            // Do nothing
-            AppLog.i(T.STATS, "StatsService is already running on this blogID - " + mServiceBlogId
-                    + " for the same period and the same date.");
         } else {
-            // stats is running on a different blogID
-            stopRefresh();
-            startTasks(blogId, period, requestedDate, updateGraphStats, updateAlltimeStats, startId);
+            requestedDate = intent.getStringExtra(ARG_DATE);
         }
-        // Always update the startId. Always.
+
+        final int maxResultsRequested = intent.getIntExtra(ARG_MAX_RESULTS, DEFAULT_NUMBER_OF_RESULTS);
+        final int pageRequested = intent.getIntExtra(ARG_PAGE_REQUESTED, -1);
+
+        StatsEndpointsEnum[] sectionsToUpdate = (StatsEndpointsEnum[]) intent.getSerializableExtra(ARG_SECTION);
```

Comment:
Not reproducible on Nexus 5 running 5.1. When I connect the debugger and inspect the intent Bundle [I see](https://cldup.com/WjV3BLVTLQ.png) the ClassCastException in place of all the data. Runs fine though, no crashes.